vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/ItemClickListenerIT.java

```diff
@@ -79,6 +80,37 @@ public void doubleClickCheckboxInCell_noEventsFired() {
         checkbox.doubleClick();
         Assert.assertEquals("", getClickMessage());
         Assert.assertEquals("", getDoubleClickMessage());
+        Assert.assertEquals("", getColumnClickMessage());
+    }
+
+    @Test
+    public void clickCell_columnNameAvailable() {
+        grid.getCell(0, 0).click();
+        Assert.assertEquals("Name", getColumnClickMessage());
+    }
+
+    @Test
+    public void doubleClickCell_columnNameAvailable() {
+        grid.getCell(0, 0).doubleClick();
+        Assert.assertEquals("Name", getColumnDoubleClickMessage());
+    }
+
+    @Test
+    public void clickDetailsCell_noColumnNameAvailable() {
```

Comment:
Great! We should also have a test case that `getItem()` works when clicking details cell. But that is not part of this feature ,so it's up to you if you want to add that test case as well.

Reply:
Oh man... Looks like I was wrong all the time. Clicking on details-cell doesn't fire the `item-click` event at all. Sorry for that. :/
It's because clicking on details-cell doesn't fire the web component's `cell-activate` event, which is used to determine the target item.
So can you change this test case:
- Change the name to `clickDetailsCell_noItemClickEventFired`
- Check also that there's no item name logged (in addition to column)

Reply:
@pekam because item click is not fired, does this mean there is no way for column information to be missing? if so, I could get rid of the `Optional` in event and just return the column

Reply:
Good point! I think so, yes. At least there _shouldn't_ be a case where the column is null.
I try to think/test if there are any corner cases where that could happen...

Reply:
Couldn't find any way to break it :)